sys/kern/src/arch/riscv32.rs

```diff
@@ -443,50 +445,43 @@ fn timer_handler() {
     crate::profiling::event_timer_isr_exit();
 }
 
-// NOTE: The kernel currently assumes that only Context 0 handles interrupts.
 pub fn disable_irq(n: u32) {
-    Plic::mask(0, n as usize);
+    let cur_mie = register::mie::read();
+    let new_mie = cur_mie.bits() & !(0x1 << n);
+    unsafe {
+        asm!("
+            csrw mie, {x}",
+            x = in(reg) new_mie,
+        );
+    }
 }
 
 pub fn enable_irq(n: u32) {
-    unsafe { Plic::unmask(0, n as usize) };
-    // Complete is called here because this tells the PLIC that the incoming
-    // interrupt has been handled. With the way the kernel has been written,
-    // there is no way for this function to be called without a task making
-    // the appropriate syscall, and in most cases that'll only happen when
-    // a task is done handling the previous interrupt.
-    Plic::complete(0, n as usize);
+    let cur_mie = register::mie::read();
+    let new_mie = cur_mie.bits() | (0x1 << n);
+    unsafe {
+        asm!("
+            csrw mie, {x}",
+            x = in(reg) new_mie,
+        );
+    }
 }
 
-// Machine external interrupt handler
+// Handler for interrupts related to the platform
 #[no_mangle]
```

Comment:
Why is this `#[no_mangle]`?

Reply:
All interrupt handlers have `#[no_mangle]`, in both ARM and RISC-V. I'm not entirely certain why ARM does it, but I'm leaning towards keeping it for now.

Reply:
In arm they are referenced by the linker, but that is not the case on riscv right now.  I think it is fine to leave as is though.